File: 250505/T를 초과하는 연속 부분 수열/subsequence-above-t.java
import java.util.Scanner;

public class Main {

    public static final int MAX_N = 1000;
    public static int[] arr = new int[MAX_N];

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int t = sc.nextInt();
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        int count = 0;
        int max = Integer.MIN_VALUE;
        for (int i = 0; i < n; i++) {
            if (i == 0) {
                if (arr[i] > t) {
                    count++;
                } else {
                    continue;
                }
            } 
            else {
                if (arr[i] > t) {
                    if (arr[i] > arr[i - 1]) {
                        count++;
                    }
                } else {
                    max = Math.max(max, count);
                    count = 1;
                }
            }
        }
        max = Math.max(max, count);
        System.out.println(max);
        sc.close();
    }
}

File: 250505/T를 초과하는 연속 부분 수열/subsequence-above-t.cpp
#include <iostream>
#include <climits>
#include <cmath>

#define MAX_N 1000

using namespace std;

int n, t;
int arr[MAX_N];

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(NULL);
    cout.tie(NULL);

    cin >> n >> t;
    for(int i = 0; i < n; i++) {
        cin >> arr[i];
    }
    int count = 0;
    int maxCount = INT_MIN;
    for(int i = 0; i < n; i++) {
        if(arr[i] > t) {
            count++;
            maxCount = max(maxCount, count);
        }
        else {
            count = 0;
        }
    }
    maxCount = max(maxCount, count);
    cout << maxCount << "\n";
    return 0;
}